### PR TITLE
Upgrade base box to Ubuntu Xenial 16.04

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

**Comment by escowles** [comment]
This doesn't build for me — `vagrant up` stops with the error:

```
==> default: Setting hostname...
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

hostname -f

Stdout from the command:



Stderr from the command:

sudo: unable to resolve host ubuntu-xenial
mesg: ttyname failed: Inappropriate ioctl for device
hostname: Name or service not known
```

**Comment by jcoyne** [comment]
Works for me. What version of VirtualBox do you have?

```
==> default: Forwarding ports...
    default: 3000 (guest) => 3000 (host) (adapter 1)
    default: 8983 (guest) => 8983 (host) (adapter 1)
    default: 8984 (guest) => 8984 (host) (adapter 1)
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: ubuntu
    default: SSH auth method: password
    default: Warning: Remote connection disconnect. Retrying...
    default:
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if it's present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
    default: The guest additions on this VM do not match the installed version of
    default: VirtualBox! In most cases this is fine, but in rare cases it can
    default: prevent things such as shared folders from working properly. If you see
    default: shared folder errors, please make sure the guest additions within the
    default: virtual machine match the version of VirtualBox you have installed on
    default: your host and reload your VM.
    default:
    default: Guest Additions Version: 5.0.24
    default: VirtualBox Version: 5.1
==> default: Setting hostname...
==> default: Mounting shared folders...
    default: /vagrant => /Users/jcoyne/workspace/hydra-vagrant
==> default: Running provisioner: shell...
    default: Running: /var/folders/9t/rygbnddx0b1ckw6tjs3m18qm0000gq/T/vagrant-shell20160926-21300-vkdvax.sh
```

**Comment by jcoyne** [comment]
@escowles also which version of Vagrant. These errors look similar to what you're seeing: https://github.com/mitchellh/vagrant/issues/5673

**Comment by jcoyne** [comment]
Our Islandora friends may have solved this: https://github.com/Islandora-CLAW/CLAW/issues/378

**Comment by escowles** [comment]
I had an older version of Vagrant (1.8.1) — I've updated Vagrant and VirtualBox to the latest version and it's getting past the initial build steps now.  I'll just wait for it to finish and double-check that fedora and the demo apps run before merging.

**Comment by escowles** [comment]
There was an error with FITS setup, which happened twice.  I was able to SSH into the vagrant VM, and noticed that `/vagrant` wasn't mounted.  So I think it was failing because the FITS download failed.  I'm rebuilding now with `master` to make sure my upgraded VirtualBox and Vagrant work.

**Comment by jcoyne** [comment]
@escowles I wonder if that's due to:

```
==> default: Checking for guest additions in VM...
    default: The guest additions on this VM do not match the installed version of
    default: VirtualBox! In most cases this is fine, but in rare cases it can
    default: prevent things such as shared folders from working properly. If you see
    default: shared folder errors, please make sure the guest additions within the
    default: virtual machine match the version of VirtualBox you have installed on
    default: your host and reload your VM.
    default:
    default: Guest Additions Version: 5.0.24
    default: VirtualBox Version: 5.1
```

**Comment by jcoyne** [comment]
@escowles I'm getting `==> default: /tmp/vagrant-shell: 19: /tmp/vagrant-shell: cannot create /home/vagrant/.bashrc: Directory nonexistent`, but I think that's because the user on this box is `ubuntu` not `vagrant`.  Is that what you are seeing?

**Comment by escowles** [comment]
Looks like it was an outdated version of the xenial base box.  After `vagrant box update`, the VM builds for me without problems.

**Comment by escowles** [comment]
@jcoyne Yes, the user is now ubuntu instead of vagrant, so that looks like a separate issue.

File: install_scripts/fits.sh [hunk 1]
```diff
@@ -7,13 +7,16 @@ if [ -f "$SHARED_DIR/install_scripts/config" ]; then
...
+  DOWNLOAD_URL="http://projects.iq.harvard.edu/files/fits/files/fits-${FITS_VERSION}.zip"
+  cd $DOWNLOAD_DIR
+  if [ ! -f "fits.zip" ]; then
+    curl $DOWNLOAD_URL -o fits.zip
+  fi
+  unzip fits.zip
+  chmod a+x fits-$FITS_VERSION/*.sh
+  FITS_PATH="${DOWNLOAD_DIR}/fits-${FITS_VERSION}"
+  cd
+  echo "PATH=\${PATH}:$FITS_PATH" >> .bashrc
```
Comment (escowles) [inline install_scripts/fits.sh:20]: This is being run as root, so I think it needs to explicitly write to `/home/ubuntu/.bashrc`.

Comment (jcoyne) [reply install_scripts/fits.sh:20]: Are you certain. Looks like it runs as ubuntu to me.  I reprovisioned and it works fine.  Are you having trouble?

Comment (escowles) [reply install_scripts/fits.sh:20]: When I rebuild from scratch, fits.sh isn't in the path of the `ubuntu` user, and the line isn't at the end of their .bashrc.

The Vagrantfile only has `privileged: false` for the `demo-app.sh` script (https://github.com/projecthydra-labs/hydra-vagrant/blob/d7d79c82ed0d484bba0a54325b81993d108cddb0/Vagrantfile#L27), so I think the others are all being run as root.

Comment (jcoyne) [reply install_scripts/fits.sh:20]: Oops, had this locally:

```
-  config.vm.provision "shell", path: "./install_scripts/fits.sh", args: shared_dir
+  config.vm.provision "shell", path: "./install_scripts/fits.sh", args: shared_dir, privileged: false
```

Comment (jcoyne) [reply install_scripts/fits.sh:20]: I pushed that change up.

Comment (escowles) [reply install_scripts/fits.sh:20]: @jcoyne That explains the difference!  This looks good to me now — I'll just build the VM one last time to be sure.

**Pull request merged**